test failure:
```
Error:  Tests run: 2, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 2.678 s <<< FAILURE! -- in org.apache.camel.upgrade.springboot.CamelSpringBoot413Test
Error:  org.apache.camel.upgrade.springboot.CamelSpringBoot413Test.yamlFile -- Time elapsed: 0.768 s <<< FAILURE!
java.lang.AssertionError: 
Expected recipe to complete in 2 cycles, but took 1 cycle. This usually indicates the recipe is making changes after it should have stabilized.
https://docs.openrewrite.org/authoring-recipes/recipe-conventions-and-best-practices#stay-single-cycle
	at org.openrewrite.test.LargeSourceSetCheckingExpectedCycles.afterCycle(LargeSourceSetCheckingExpectedCycles.java:121)
	at org.openrewrite.RecipeScheduler.runRecipeCycles(RecipeScheduler.java:100)
	at org.openrewrite.RecipeScheduler.scheduleRun(RecipeScheduler.java:42)
	at org.openrewrite.Recipe.run(Recipe.java:447)
	at org.openrewrite.test.RewriteTest.rewriteRun(RewriteTest.java:415)
	at org.openrewrite.test.RewriteTest.rewriteRun(RewriteTest.java:134)
	at org.apache.camel.upgrade.springboot.CamelSpringBoot413Test.yamlFile(CamelSpringBoot413Test.java:62)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
```

--- a/camel-spring-boot-upgrade-recipes/src/test/java/org/apache/camel/upgrade/springboot/CamelSpringBoot413Test.java
+++ b/camel-spring-boot-upgrade-recipes/src/test/java/org/apache/camel/upgrade/springboot/CamelSpringBoot413Test.java
@@ -82,7 +82,7 @@ class CamelSpringBoot413Test implements RewriteTest {
                             name: 'Foo'
                           routecontroller.backOffMultiplier: 5
                           routecontroller.backOffDelay: true
-                          main.runController: something
+                          main.run-controller: something
                         another:
                           ignored:
                             property: true

--- a/camel-spring-boot-upgrade-recipes/src/test/java/org/apache/camel/upgrade/springboot/CamelSpringBoot413Test.java
+++ b/camel-spring-boot-upgrade-recipes/src/test/java/org/apache/camel/upgrade/springboot/CamelSpringBoot413Test.java
@@ -60,9 +60,9 @@ class CamelSpringBoot413Test implements RewriteTest {
     @Test
     void yamlFile() {
         rewriteRun(
-                //both org.apache.camel.upgrade.camel413.CamelSpringBootMigrationRecipe and
-                // org.apache.camel.upgrade.UpdateCamelSpringBootPropertiesAndYamlKeys modifies this properties
-                spec -> spec.expectedCyclesThatMakeChanges(2),
+                //only org.apache.camel.upgrade.UpdateCamelSpringBootPropertiesAndYamlKeys modifies this file;
+                // YamlDsl413Recipe is gated on Camel YAML DSL root keys and does not apply to Spring Boot application.yaml
+                spec -> spec.expectedCyclesThatMakeChanges(1),
             yaml(
                     """
                       camel:

